sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/SqlRuleFilter.java

```diff
@@ -99,4 +100,46 @@ Boolean isPreprocessingRequired() {
     public String getSqlExpression() {
         return sqlExpression;
     }
+
+    /**
+     * Converts the value of the current instance to its equivalent string representation.
+     *
+     * @return A string representation of the current instance.
+     */
+    @Override
+    public String toString() {
+        return String.format("SqlRuleFilter: %s", sqlExpression);
+    }
+
+    /**
+     *  Compares this RuleFilter to the specified object. The result is true if and only if the argument is not null
+     *  and is a SqlRuleFilter object that with the same parameters as this object.
+     *
+     * @param other - the object to which the current SqlRuleFilter should be compared.
+     * @return True, if the passed object is a SqlRuleFilter with the same parameter values, False otherwise.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (!(other instanceof SqlRuleFilter)) {
+            return false;
+        }
+        SqlRuleFilter that = (SqlRuleFilter) other;
+        return sqlExpression.equals(that.sqlExpression)
+            && Objects.equals(compatibilityLevel, that.compatibilityLevel)
+            && Objects.equals(requiresPreprocessing, that.requiresPreprocessing)
+            && Objects.equals(properties, that.properties);
+    }
+
+    /**
+     * Returns a hash code for this SqlRuleFilter, which is the hashcode for the SqlExpression.
+     *
+     * @return a hash code value for this object.
+     */
+    @Override
+    public int hashCode() {
+        return sqlExpression.hashCode();
```

Comment:
Is there any reason that we only use `sqlExpression` in hashCode, but equals contains all fields (nullable field seems acceptable for `Objects.hash`)?

Reply:
In my original pull request, I did hash the three attributes.  My submission implements this differently

![image](https://user-images.githubusercontent.com/1204224/159675405-5675fa5f-31ad-4417-a5d4-7cbb0ef769af.png)

Reply:
The idea behind adding just the sql expression into the hashcode here is that sql expression is the important unique identifier of the filter (the only one settable by client), and is sufficient to produce a unique hashcode. 

Here is the C# code as well, following the same idea - https://github.com/Azure/azure-sdk-for-net/blob/0bda2178d9677f156f80f0793998a6a99edf93e7/sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/Rules/SqlRuleFilter.cs#L66

Reply:
It does make sense, that these objects are consistent with the C# implementation.

I am trying to reconcile the accepted practice that objects that are equal (via the equals method) have the same hashCode.  This will still be true, but then we will have objects that are not equal which still have the same hashCode (two SqlExpression filters that have the same sqlExpression but different compatibility levels, for example).

The safest path could have the equals and hashCode methods implement using the same attributes.  But, this could be splitting hairs. :)  More thinking out loud.